Infrastructure.DataAccess/Migrations/SQLScripts/Migrate_To_Data_Processing_Registration.sql

```diff
@@ -0,0 +1,656 @@
+*/
+
+BEGIN
+
+	DECLARE @MigrationContext1 TABLE 
+    (
+		rowNumber int IDENTITY(1,1) PRIMARY KEY,
+        ItContractId int,
+		ItSystemUsageId int,
+		OrganizationId int,
+		LastChanged datetime2(7),
+		Name varchar(max),
+		AgreementConcluded int, 
+		AgreementConcludedUrl varchar(max),
+		AgreementConcludedUrlName varchar(max),
+		DataResponsibleNote varchar(max),
+		OversightConcluded int,
+		OversightConcludedDate varchar(max),
+		OversighConcludedRemark varchar(max),
+		DatahandlerSupervisionLink varchar(max),
+		DatahandlerSupervisionLinkName varchar(max),
+		ContractUserId int,
+		ContractLastChangedBy int,
+		SystemUsageUserId int
+    )
+	
+
+	/*
+		Get all info needed for first migration situation
+	*/
+
+	INSERT INTO @MigrationContext1 (ItContractId, ItSystemUsageId, OrganizationId, LastChanged, Name, AgreementConcluded, AgreementConcludedUrl, AgreementConcludedUrlName, DataResponsibleNote, OversightConcluded, OversightConcludedDate, OversighConcludedRemark, DatahandlerSupervisionLink, DatahandlerSupervisionLinkName, ContractUserId, ContractLastChangedBy, SystemUsageUserId)
+        SELECT
+			ItContractItSystemUsages.ItContractId, 
+			ItContractItSystemUsages.ItSystemUsageId,
+			ItContract.OrganizationId,
+			GETUTCDATE(),
+			'Konverteret_Databehandling_',
+			ItContract.ContainsDataHandlerAgreement,
+			ItContract.DataHandlerAgreementUrl,
+			ItContract.DataHandlerAgreementUrlName,
+			ItSystemUsage.dataProcessor,
+			CASE
+				WHEN ItSystemUsage.dataProcessorControl = 0 THEN 1
+				WHEN ItSystemUsage.dataProcessorControl = 1 THEN 0
+				WHEN ItSystemUsage.dataProcessorControl IS NULL THEN NULL
+				ELSE 2
+			END AS OversightConcluded,
+			ItSystemUsage.lastControl,
+			ItSystemUsage.noteUsage,
+			ItSystemUsage.datahandlerSupervisionDocumentationUrl,
+			ItSystemUsage.datahandlerSupervisionDocumentationUrlName,
+			ItContract.ObjectOwnerId,
+			ItContract.LastChangedByUserId,
+			ItSystemUsage.ObjectOwnerId
+        FROM 
+		ItContractItSystemUsages 
+			INNER JOIN ItContract ON ItContract.Id = ItContractItSystemUsages.ItContractId
+			INNER JOIN ItSystemUsage ON ItSystemUsage.Id = ItContractItSystemUsages.ItSystemUsageId
+		WHERE 
+			ItContractItSystemUsages.ItContractId 
+				IN (
+					SELECT 
+						Id 
+					FROM 
+						ItContract 
+					WHERE 
+						ContractTypeId = 5
+				)
+
+	DECLARE @DprIds1 table (rowNumber int IDENTITY(1,1) PRIMARY KEY, id int)
+
+	/*
+		Create new DPR's and save created Id's in a new table
+	*/
+
+	INSERT INTO 
+		DataProcessingRegistrations (OrganizationId, Name, LastChanged, IsAgreementConcluded, DataResponsibleRemark, IsOversightCompleted, LatestOversightDate, OversightCompletedRemark, ObjectOwnerId, LastChangedByUserId)
+	OUTPUT
+		inserted.Id into @DprIds1
+	SELECT
+		OrganizationId, Name, LastChanged, AgreementConcluded, DataResponsibleNote, OversightConcluded, OversightConcludedDate, OversighConcludedRemark, ContractUserId, ContractLastChangedBy
+	FROM 
+		@MigrationContext1 
```

Comment:
Oprettes fra context tabellen så "rækkenummer" skulle gerne matche mellem de to tabeller. Men er der garanti for det?